src/Dapr.AspNetCore/CloudEventsMiddleware.cs

```diff
@@ -148,16 +189,55 @@ private async Task ProcessBodyAsync(HttpContext httpContext, string charSet)
             }
         }
 
-        private string GetDataContentType(JsonElement json, out bool isJson)
+        private void ForwardCloudEventPropertiesAsHeaders(
+            HttpContext httpContext,
+            IEnumerable<JsonProperty> jsonPropNames)
         {
+            if (!options.ForwardCloudEventPropertiesAsHeaders)
+            {
+                return;
+            }
+
+            var filteredPropertyNames = jsonPropNames
+                .Where(d => !ExcludedPropertiesFromHeaders.Contains(d.Name, StringComparer.OrdinalIgnoreCase));
+
+            if (options.IncludedCloudEventPropertiesAsHeaders != null)
+            {
+                filteredPropertyNames = filteredPropertyNames
+                    .Where(d => options.IncludedCloudEventPropertiesAsHeaders
+                        .Contains(d.Name, StringComparer.OrdinalIgnoreCase));
+            }
+            else if (options.ExcludedCloudEventPropertiesFromHeaders != null)
+            {
+                filteredPropertyNames = filteredPropertyNames
+                    .Where(d => !options.ExcludedCloudEventPropertiesFromHeaders
+                        .Contains(d.Name, StringComparer.OrdinalIgnoreCase));
+            }
+
+            foreach (var jsonProperty in filteredPropertyNames)
+            {
+                httpContext.Request.Headers.TryAdd($"Cloudevent.{jsonProperty.Name.ToLowerInvariant()}",
+                    jsonProperty.Value.GetRawText().Trim('\"'));
+            }
+        }
+
+        private static string GetDataContentType(JsonElement json, out bool isJson)
+        {
+            var dataContentTypePropName = json.EnumerateObject()
+                                              .Select(d => d.Name)
+                                              .FirstOrDefault(d =>
+                                                  d.Equals(CloudEventPropertyNames.DataContentType,
+                                                      StringComparison.OrdinalIgnoreCase))
+                                          ?? "";
```

Comment:
Similar nit here as before; I don't see a need to use `""` as a default; just test whether `dataContentTypePropName` is non-`null` before calling `TryGetProperty()` in the following `if` statement.

Reply:
On this one I agree, we don't want to pick the wrong JSON field, must have missed it when I changed the other ones above.
I'll change it to use `string.IsNullOrWhiteSpace` first thing in the if, based on the argument in the above comment.